app/vmselect/promql/exec.go

```diff
@@ -13,10 +13,81 @@ import (
 	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
 	"github.com/VictoriaMetrics/metrics"
 	"github.com/VictoriaMetrics/metricsql"
+
+	"github.com/google/uuid"
 )
 
 var logSlowQueryDuration = flag.Duration("search.logSlowQueryDuration", 5*time.Second, "Log queries with execution time exceeding this value. Zero disables slow query logging")
 
+type Query struct {
+	q       string
+	ec      *EvalConfig
+	stopCh  chan error
+	startAt time.Time
+}
+
+type queriesMap struct {
+	mu sync.Mutex
+	m  map[string]Query
+	c  int64
+}
+
+func (rqm *queriesMap) Init() {
+	rqm.m = make(map[string]Query)
+}
+
+func (rqm *queriesMap) Add(q Query) string {
+	rqm.mu.Lock()
+	c := uuid.New().String()
```

Comment:
agree with that, but using `uint64` may cause some issues when restarting the vmselect instance(accidentally kill the unexpected query since the qid is uint64 and auto increment)

Reply:
The probability of `queryID` clash after `vmselect` restart may be reduced to 0 by initializing `nextQueryID` to the current time in nanoseconds - see https://github.com/VictoriaMetrics/VictoriaMetrics/blob/2d9b3ad5b386dd42ec67e42b8f44a5efae750b4d/app/vmselect/promql/active_queries.go#L80 . This prevents from `queryID` clash after `vmselect` restart if incoming queries' rate is smaller than a billion quries per second (such a rate is impossible).

Of course, the `queryID` generated via `atomic.AddUint64` is easy to predict. But why do you think `queryID` must be unpredictable? Which issue does this solve?